src/sqlfluff/core/linter/linter.py

```diff
@@ -267,7 +272,21 @@ def parse_noqa(comment: str, line_no: int):
                             )
                     rules: Optional[Tuple[str, ...]]
                     if rule_part != "all":
-                        rules = tuple(r.strip() for r in rule_part.split(","))
+                        # Rules can be globs therefore we compare to the rule_set to expand the globs.
+                        unexpanded_rules = tuple(
+                            r.strip() for r in rule_part.split(",")
+                        )
+                        possible_rule_codes = [r.code for r in rule_set]
+                        expanded_rules = []
```

Comment:
Is it possible we'll end up with multiple occurrences of the same rule? If so, should we filter out duplicates before returning?

E.g. `L0*5` and `L01*` will both match "L015".

Reply:
It's possible, but I thought since we later just check if the raised rule is `in` the rules list it wasn't a big issue. However, I think you're right in terms of a design decision as it means the results from this function are better defined 👍

Reply:
Cool. One reason I think this is worth avoiding (i.e. fixing) is that it could somehow cause the same rule to be run twice -- if not today, maybe in a future version.

Reply:
makes sense, I've added an added some logic to remove dups and a test rule to verify it 👍